org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/models/DeviceModel.java

```diff
@@ -30,20 +30,29 @@
 		UNBOXEDTYPES.put(Short.class, short.class);
 	}
 	
-	private String type;
+	private String name, type;
 	private Map<String, Object> deviceConfiguration;
 	private List<DeviceModel> roiConfiguration;
 	
 	public DeviceModel(String type, Map<String, Object> deviceConfiguration) {
-		this(type, deviceConfiguration, new ArrayList<>());
+		this(type, deviceConfiguration, null);
 	}
 	
 	public DeviceModel(String type, Map<String, Object> deviceConfiguration, List<DeviceModel> roiConfiguration) {
 		this.type = type;
 		this.deviceConfiguration = deviceConfiguration;
+		if (roiConfiguration == null) roiConfiguration = new ArrayList<>();//NPE protection

```

Comment:
Is this what we want? Could have Objects.requireNonNull(roiConfiguration) so well will get a NPE thrown here is it useful to instantiate a DeviceModel will null?